File: app/graphql/types/custom/author_type.rb
module Types
  module Custom
    class AuthorType < Types::BaseObject
      field :id, Integer, null: false
      field :name, String, null: true
      field :logo, String, null: true
      field :created_at, GraphQL::Types::ISO8601DateTime, null: false
      field :updated_at, GraphQL::Types::ISO8601DateTime, null: false

      # Custom Fields
      field :totalViews, resolver: Resolvers::ResolversAuthor::Fields::FieldTotalViews, null: false
    end
  end
end

File: angular/src/app/types/graphql/author-type.ts
import * as moment from "moment-timezone";
import {AuthorAnalyticsType} from "./author-analytics-type";

export interface AuthorType {
  id: number;
  logo: string | null;
  name: string | null;
  createdAt: moment.Moment;
  updatedAt: moment.Moment;

  // Custom Fields
  analytics: AuthorAnalyticsType | null;
}
